### ISSUE TITLE
Issue : Job ran at 11:59 pm instead of 12:00 am (cron : */30 * * * *)

### ISSUE THREAD

**Issue by Gunner17** [node-schedule/node-schedule#440]
I am using node-schedule version 1.2.3
I have scheduled a Job using the following cron expression.
 */30 * * * *
This job is supposed to run at every 30 minutes. 
But it ran at 11:59 pm instead of 12:00 am, any particular reason for that?
(04-May-2018 11:59:59.941 pm to be exact)

I don't mind the job running few milliseconds/seconds later than the scheduled time, but why would it run before time?

Then, the next job ran at 12:35 am, instead of 12:30 am

Do I need to create the jobs using recurrence rules instead of the below Cron expressions ?
1. */30 * * * *
2. 10-23 * * *

Or is this a know issue with 1.2.3 version? 
If I use 1.3.0 will this issue go away?

**Comment by santigimeno** [comment]
Yeah, I think that was solved by: https://github.com/node-schedule/node-schedule/commit/28fa70a94dd17ef84fbc627aa7c8f3111707f1d4. Can you test with `1.3.0`?

**Comment by fritx** [comment]
+1 here.

```plain
start=2018-05-23 21:59:43.019
instead of 22:00
```

**Comment by santigimeno** [comment]
@fritx what version are you using? If it's the latest, do you have reproducible code?

**Comment by fritx** [comment]
@santigimeno sorry for that, I was using `1.2.3`. I've just upgraded it, thanks

**Comment by fritx** [comment]
Btw, how could this reproduce? Is it a JS engine bug?
